From 18bf98457aa55468fc9800eab5b4f78f95f546c9 Mon Sep 17 00:00:00 2001
From: Mat Booth <mat.booth@redhat.com>
Date: Thu, 20 Feb 2020 16:08:00 +0000
Subject: [PATCH 5/8] Fix uncaught exception

"unreported exception org.codehaus.plexus.configuration.PlexusConfigurationException; must be caught or declared to be thrown"
---
 .../tycho/extras/sourcefeature/SourceFeatureMojo.java      | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tycho-extras/tycho-source-feature-plugin/src/main/java/org/eclipse/tycho/extras/sourcefeature/SourceFeatureMojo.java b/tycho-extras/tycho-source-feature-plugin/src/main/java/org/eclipse/tycho/extras/sourcefeature/SourceFeatureMojo.java
index 9c95f2d..1f34a82 100644
--- a/tycho-extras/tycho-source-feature-plugin/src/main/java/org/eclipse/tycho/extras/sourcefeature/SourceFeatureMojo.java
+++ b/tycho-extras/tycho-source-feature-plugin/src/main/java/org/eclipse/tycho/extras/sourcefeature/SourceFeatureMojo.java
@@ -560,7 +560,12 @@ public class SourceFeatureMojo extends AbstractMojo {
     }
 
     private String getAttribute(PlexusConfiguration dom, String attrName) {
-        String attr = dom.getAttribute(attrName);
+        String attr = null;
+        try {
+            attr = dom.getAttribute(attrName);
+        } catch (Exception ex) {
+            attr = null;
+        }
         if (attr == null) {
             return null;
         }
-- 
2.21.1

